components/builder-api-client/src/artifactory.rs

```diff
@@ -351,30 +361,27 @@ impl BuilderAPIProvider for ArtifactoryClient {
                       -> Result<()> {
         let path = self.url_path_for_key(origin, revision);
 
-        let mut file =
+        let file =
             File::open(src_path).map_err(|e| Error::KeyReadError(src_path.to_path_buf(), e))?;
         let file_size = file.metadata()
                             .map_err(|e| Error::KeyReadError(src_path.to_path_buf(), e))?
                             .len();
 
-        let result = if let Some(mut progress) = progress {
+        let resp = if let Some(mut progress) = progress {
             progress.size(file_size);
-            let mut reader = TeeReader::new(file, progress);
-            self.add_authz(self.0.put(&path), token)
-                .body(Body::SizedBody(&mut reader, file_size))
-                .send()
+            let reader = TeeReader::new(file, progress);
+            let body = Body::sized(reader, file_size);
+            self.add_authz(self.0.put(&path), token).body(body).send()?
         } else {
-            self.add_authz(self.0.put(&path), token)
-                .body(Body::SizedBody(&mut file, file_size))
-                .send()
+            let body = Body::sized(file, file_size);
+            self.add_authz(self.0.put(&path), token).body(body).send()?
```

Comment:
Good catch on the duplicated code - will fix.

Re: the result/response - I'm updating so that Result = "res" and Response = "resp" in accordance with the Reqwest samples so we are consistent in the usage.